### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-shubham-morankar**
PR Title: https://pmdb.fortinet.com/ProjectManagement/viewDocument.php?id=22390&doc_type=design

Problem Statement:

The SCADAfence Platform is an OT/ICS cybersecurity solution. It integrates alerts from the solution to the QRadar feed.

Requested Internal:
• Create Alert
• Get Alert List
• Update Alert Status
• Get Asset List
• Update Asset
• Get Sites Status

Description: Developed new connector to integrates alerts, assets ans sites from the SCADAfence.

Impact: None

Unit Testcase:

1. Install/Upgrade the Connector
2. Check dependency packages are install if required
3. Configure connector using (SERVER URL, Account Route, API Key, and SECRET Key)
4. Check all the playbooks are present
5. Check all the playbooks tags are present
6. Check all the playbooks should be in Info mode
7. Connector version and Sample playbook version should be same

### CODE REVIEW

File: scadafence/info.json [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+        },
+        {
+          "name": "severity",
+          "title": "Alert Severity",
+          "description": "Select the severity of alert based on which you want to create alert in SCADAfence. You can choose from the following options: Information, Warning, Threat, Severe, Critical, or Critical.",
+          "type": "select",
+          "visible": true,
+          "editable": true,
+          "required": false,
+          "options": [
```
Comment (cs-pranjali-munfan) [inline scadafence/info.json:92]: Add default value for severity

File: scadafence/info.json [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+          ]
+        },
+        {
+          "name": "details",
+          "title": "Description",
+          "description": "Specify the description of the alert based on which you want to create alert in SCADAfence.",
+          "type": "text",
+          "visible": true,
+          "editable": true,
+          "required": true,
```
Comment (cs-pranjali-munfan) [inline scadafence/info.json:108]: Is this description a required parameter?

File: scadafence/info.json [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+          "title": "Remediation Text",
+          "description": "Specify the remediation text based on which you want to create alert in SCADAfence.",
+          "type": "text",
+          "visible": true,
+          "editable": true,
+          "required": false,
+          "tooltip": "Specify the remediation text based on which you want to create alert in SCADAfence."
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-pranjali-munfan) [inline scadafence/info.json:133]: If possible, add output schema

File: scadafence/info.json [hunk 4]
```diff
@@ -0,0 +1,737 @@
...
+          "name": "order",
+          "title": "Order By",
+          "description": "Select the order by of alerts based on which you want to filter the alerts retrieved from SCADAfence. You can choose from the following options: Severity or Site ID",
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "select",
+          "tooltip": "Select the order by of alerts based on which you want to filter the alerts retrieved from SCADAfence. You can choose from the following options: Severity or Site ID",
+          "options": [
+            "Severity",
```
Comment (cs-pranjali-munfan) [inline scadafence/info.json:264]: Add default value

File: scadafence/info.json [hunk 5]
```diff
@@ -0,0 +1,737 @@
...
+          "editable": true,
+          "required": true,
+          "options": [
+            "Created",
+            "InProgress",
+            "Resolved"
+          ]
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-pranjali-munfan) [inline scadafence/info.json:359]: Add output schema

File: scadafence/operations.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    '413': 'Request Entity Too Large, Sample file size over max limit',
+    '415': 'Unsupported Media Type',
+    '418': 'Unsupported File Type Sample, file type is not supported',
+    '419': 'Request quota exceeded',
+    '420': 'Insufficient arguments',
+    '421': 'Invalid arguments',
+    '500': 'Internal error',
+    '502': 'Bad Gateway',
+    '513': 'File upload failed'
+}
```
Comment (cs-pranjali-munfan) [inline scadafence/operations.py]: add this in constant.py file

File: scadafence/operations.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+
+def create_alert(config, params):
+    sf = SCADAfence(config)
+    endpoint = '/alert'
+    try:
+        payload = check_payload(params)
+        response = sf.make_rest_call(endpoint, 'POST', data=json.dumps(payload))
+        return response
+    except Exception as err:
```
Comment (cs-suyog-jinturkar) [inline scadafence/operations.py]: Why we need exception handling here? "make_rest_call" already has it in place. Can you please elaborate this? @cs-shubham-morankar

Comment (cs-suyog-jinturkar) [reply scadafence/operations.py]: Same applies to all other actions

Comment (cs-shubham-morankar) [reply scadafence/operations.py]: Yes Correct, we already handled in make_rest_call. Hence, removed all the exception handling from the actions.

Comment (cs-suyog-jinturkar) [reply scadafence/operations.py]: Okey..

**Review by cs-pranjali-munfan — APPROVED** [review]
Looks good to me

**Pull request merged**